climatedata_api/map.py

```diff
@@ -241,3 +243,59 @@ def get_id_list_from_points(compressed_points):
             break
 
     return gids.tolist()
+
+def get_s2d_release_date(var, freq):
+    """
+    Return the release date of the forecast data associated to a given variable and frequency
+    curl 'http://localhost:5000/get-s2d-release-date/air_temp/seasonal'
+    """
+    try:
+        if var not in app.config['S2D_VARIABLES']:
+            raise ValueError
+        if freq not in app.config['S2D_FREQUENCIES']:
+            raise ValueError
+    except ValueError:
+        return "Bad request", 400
+
+
+    dataset = open_dataset_by_path(app.config['NETCDF_S2D_FORECAST_FILENAME_FORMATS'].format(
+        root=app.config['DATASETS_ROOT'],
+        var=var,
+        freq=freq
+    ))
+
+    latest_datetime = dataset['time'].min().values
+    latest_datetime_str = str(latest_datetime.astype('datetime64[D]'))
```

Comment:
Le résultat de `str()` dépend de si le datetime contient une heure ou pas.

```python
str(np.datetime64('2025-10-05'))
>>> '2025-10-05'

str(np.datetime64('2025-10-05 15:35:12'))
>>> '2025-10-05T15:35:12'
```

Il faudrait donc s'assurer que le format soit uniforme. Je pense que ce qui suit peut fonctionner (à valider en mettant les tests à jour):

```suggestion
    latest_datetime_str = latest_datetime.item().strftime('%Y-%m-%d')
```

Reply:
Le code actuel me semble correct :

```
>>> str(np.datetime64('2025-10-05').astype('datetime64[D]'))
'2025-10-05'
>>> str(np.datetime64('2025-10-05 15:35:12').astype('datetime64[D]'))
'2025-10-05'
>>> np.datetime64('2025-10-05 15:35:12').astype('datetime64[D]')
numpy.datetime64('2025-10-05')
>>> np.datetime64('2025-10').astype('datetime64[D]')
numpy.datetime64('2025-10-01')
```

Le `D` de `astype('datetime64[D]')` devrait déjà uniformiser les dates pour avoir le format YYYY-MM-DD.

Reply:
Parfait dans ce cas